pkg/manifest/os.go

```diff
@@ -881,6 +881,29 @@ func prependKernelCmdlineStage(pipeline osbuild.Pipeline, kernelOptions []string
 		)
 	}
 
+	if mountUnits {
+		// The systemd-remount-fs service reads /etc/fstab to discover mount
+		// options for / and /usr. Without an /etc/fstab, / and /usr do not get
+		// remounted, which means if they are mounted read-only in the initrd,
+		// they will remain read-only. Flip the option if we're using only
+		// mount units, otherwise the filesystems will stay mounted 'ro'.
+		//
+		// See https://www.freedesktop.org/software/systemd/man/latest/systemd-remount-fs.service.html
+		for idx := range kernelOptions {
+			if kernelOptions[idx] == "ro" {
+				kernelOptions[idx] = "rw"
+				break
+			}
+		}
+
+		// set the rootflags for the same reason as above
+		fsOptions, err := rootFs.GetFSTabOptions()
+		if err != nil {
+			panic(fmt.Sprintf("error getting filesystem options for / mountpoint: %s", err))
+		}
+		kernelOptions = append(kernelOptions, fmt.Sprintf("rootflags=%s", fsOptions.MntOps))
```

Comment:
Cosmetic: Since *most* use cases don't use `rootflags` it'd be nice to not end up with `rootflags=default`, so we could special case that and not emit `rootflags` at all.

Reply:
Yes, this is good advice.  We also end up with duplicate rootflags when the root is on btrfs or luks.  I'll clean it up.